subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/ZipHasher.java

```diff
@@ -157,6 +142,50 @@ private void fingerprintZipEntry(ZipEntry zipEntry, String fullName, List<FileSy
         }
     }
 
+    private void fingerprintManifestZipEntry(String rootParentName, List<FileSystemLocationFingerprint> fingerprints, ZipEntry zipEntry, String fullName) throws IOException {
+        byte[] manifestBytes = ByteStreams.toByteArray(zipEntry.getInputStream());
```

Comment:
Can you please add a comment that clarifies *why* the entry is copied to the byte array?